### PR TITLE
Move local corefx targets/tasks to BuildTools

### PR DESCRIPTION
**Pull request by weshaggard**
This change mostly deals with moving Tools-Override and CoreFxTools
task library to BuildTools and consuming them from there instead.

We should not merge this until the associated BuildTools PR is merged. https://github.com/dotnet/buildtools/pull/1460. I did publish a private package for this BuildTools version to test it but we shouldn't merge with that private buildtools being referenced.

### CODE REVIEW

**Comment by weshaggard** [comment]
@ericstj see the changes to build.proj to see the configuration changes. I just generated them in a target there. I moved the configuration props file into a new root level folder called "tools-local" I called it that as we have plans to move other tools/scripts/targets/props file into that directory in the future so I thought I would just see it with those.

File: buildvertical.targets [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  <UsingTask TaskName="FindBestConfigurations" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
   <PropertyGroup>
     <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
     <_traversalBuildConfigurations>$(BuildConfiguration)</_traversalBuildConfigurations>
     <_traversalBuildConfigurations Condition="'$(AdditionalBuildConfigurations)' != ''">$(_traversalBuildConfigurations);$(AdditionalBuildConfigurations)</_traversalBuildConfigurations>
   </PropertyGroup>
 
   <!-- Import configuration data model -->
-  <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
+  <Import Project="$(ProjectDir)tools-local/properties.props" />
```
Comment (ericstj) [inline buildvertical.targets]: Looks good.

**Comment by ericstj** [comment]
Looks like you might be missing a change I merged recently that added [Output] to SaveItems.Files.

**Comment by mellinoe** [comment]
nit: Should "tools-local" be "Tools-Local" for consistency with "Tools"?

**Comment by weshaggard** [comment]
>nit: Should "tools-local" be "Tools-Local" for consistency with "Tools"?

I thought about that but for things that are checked I much prefer all lowercase. I would change Tools as well if it wasn't such a large change.

**Comment by weshaggard** [comment]
Ready to merge once green.

**Comment by weshaggard** [comment]
@mellinoe @mmitche while looking at the logs for this PR I noticed an existing issue which is that the step init-tools is talking >30 minutes on these CI machines which I have no idea why. The same init-tools finishes quickly in other repo's like standard, or locally, or official builds.

https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_release_prtest/7085/consoleFull
```
20:51:11 Installing dotnet cli...
20:51:20 Restoring BuildTools version 2.0.0-prerelease-01528-02...
21:22:45 Initializing BuildTools...
21:23:47 Done initializing tools.
```

The Restoring step seems to take about 30 minutes for some reason. To help diagnose this I added the init-tools.log file to the archived files to see if anything looks off. Here is an example one:
https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_release_prtest/7085/artifact/init-tools.log

It said the restore took `Restore completed in 876.15 ms for D:\j\workspace\windows_nt_re---37265eab\init-tools.msbuild.`, so I have no idea why this step just seems to hang. It seems to do it not only on windows but the others as well. If we can figure out what is going on here our CI legs will reduce from 45 mins to 15 mins which will take a ton of pressure off the CI system. I don't know what else to really try to  get further information @mmitche should we take a machine offline and test it locally?

**Comment by weshaggard** [comment]
For reference see this log in standard:
https://ci.dot.net/job/dotnet_standard/job/master/job/Windows_NT_prtest/76/console
```
19:08:28 Installing dotnet cli...
19:08:36 Restoring BuildTools version 2.0.0-prerelease-01528-02...
19:08:39 Initializing BuildTools...
19:09:08 Done initializing tools.
```

It is installing the same CLI and restoring the same build tools package and completes in 3 seconds.

**Comment by mmitche** [comment]
@weshaggard The machine is gone now, but I saw an unrelated machine do something very similar (not 30 mins, but at least a few) on another ci system recently.

**Comment by weshaggard** [comment]
@mmitche it seems to be happening on every PR leg I've looked at. Another random case I see:

https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_release_prtest/7091/consoleFull
```
02:37:56 Installing dotnet cli...
02:38:04 Restoring BuildTools version 2.0.0-prerelease-01528-02...
03:10:15 Initializing BuildTools...
03:10:54 Done initializing tools.
```
I would really like to get to the bottom of what is happening here.

**Pull request merged**